tests/acceptance/general/ProductTagCest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Tests that subscribers are added to ConvertKit and tagged
+ * based on the product purchased.
+ *
+ * @since   1.2.0
+ */
+class ProductTagCest
+{
+	/**
+	 * Run common actions before running the test functions in this class.
+	 *
+	 * @since   1.2.0
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function _before(AcceptanceTester $I)
+	{
+		// Activate Plugins.
+		$I->activateConvertKitPlugin($I);
+		$I->activateThirdPartyPlugin($I, 'membermouse-platform');
+	}
+
+	/**
+	 * Test that the member is tagged with the configured "apply tag on add"
+	 * setting when added to a Membership Level.
+	 *
+	 * @since   1.2.0
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testMemberTaggedWhenProductPurchased(AcceptanceTester $I)
+	{
+		// Create a product.
+		$productReferenceKey = 'pTRLc9';
+		$productID           = $I->memberMouseCreateProduct($I, 'Product', $productReferenceKey);
+
+		// Setup Plugin to tag users purchasing the product to the
+		// ConvertKit Tag ID.
+		$I->setupConvertKitPlugin(
+			$I,
+			[
+				'convertkit-mapping-product-1' => $_ENV['CONVERTKIT_API_TAG_ID'],
+			]
+		);
+
+		// Generate email address for test.
+		$emailAddress = $I->generateEmailAddress();
+
+		// Enable test mode for payments.
+		$I->amOnAdminPage('admin.php?page=payment_settings');
+		$I->checkOption('test_payment_service_enabled');
+		$I->click('Save Payment Methods');
+		$I->wait(5);
+		$I->acceptPopup();
+
+		// Logout.
+		// We don't use logOut() as MemberMouse hijacks the logout process with a redirect,
+		// resulting in the logOut() assertion `loggedout=true` failing.
+		$I->amOnPage('wp-login.php?action=logout');
+		$I->click("//a[contains(@href,'action=logout')]");
+
+		// Navigate to purchase screen for the product.
+		$I->amOnPage('checkout/?rid=' . $productReferenceKey);
+
+		// Complete checkout.
+		$I->fillField('mm_field_first_name', 'First');
+		$I->fillField('mm_field_last_name', 'Last');
+		$I->fillField('mm_field_email', $emailAddress);
+		$I->fillField('mm_field_password', '12345678');
+		$I->fillField('mm_field_phone', '12345678');
+		$I->fillField('mm_field_cc_number', '4242424242424242');
+		$I->fillField('mm_field_cc_cvv', '123');
+		$I->selectOption('mm_field_cc_exp_year', '2038');
+		$I->fillField('mm_field_billing_address', '123 Main Street');
+		$I->fillField('mm_field_billing_city', 'Nashville');
+		$I->selectOption('#mm_field_billing_state_dd', 'Tennessee');
```

Comment:
Just verifying the octothorp/hash mark is intended.

Reply:
Yes, this is intended. The field doesn't have a `name` attribute in MemberMouse, for some reason.